src/core/types/execution_context.ts

```diff
@@ -7,7 +7,9 @@
  */
 
 /** @public */
-export interface KibanaExecutionContext {
+// to make it compatible with SerializableState
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type KibanaExecutionContext = {
```

Comment:
I used to get `is incompatible with index signature` error when declared it as `interface`.
We cannot use `KibanaExecutionContext extends SerializableState` since Core cannot depend on the Kibana plugins.

Reply:
i think this is the relevant typescript issue: https://github.com/microsoft/TypeScript/issues/15300